contracts/misc/TimeLock.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IERC20} from "../dependencies/openzeppelin/contracts/IERC20.sol";
+import {IERC721} from "../dependencies/openzeppelin/contracts/IERC721.sol";
+import {IERC1155} from "../dependencies/openzeppelin/contracts/IERC1155.sol";
+import "../dependencies/openzeppelin/upgradeability/OwnableUpgradeable.sol";
+import "../dependencies/openzeppelin/upgradeability/ReentrancyGuardUpgradeable.sol";
+import {EnumerableSet} from "../dependencies/openzeppelin/contracts/EnumerableSet.sol";
+import {ITimeLock} from "../interfaces/ITimeLock.sol";
+import {IPool} from "../interfaces/IPool.sol";
+import {DataTypes} from "../protocol/libraries/types/DataTypes.sol";
+import {GPv2SafeERC20} from "../dependencies/gnosis/contracts/GPv2SafeERC20.sol";
+
+contract TimeLock is ITimeLock, OwnableUpgradeable, ReentrancyGuardUpgradeable {
+    using GPv2SafeERC20 for IERC20;
+
+    struct Agreement {
+        DataTypes.AssetType assetType;
+        bool isFrozen;
+        address asset;
+        address beneficiary;
+        uint256 releaseTime; //TODO change to smaller unit
+        uint256[] tokenIdsOrAmounts;
+    }
+
+    event AgreementCreated(
+        uint256 agreementId,
+        DataTypes.AssetType assetType,
+        address indexed asset,
+        uint256[] tokenIdsOrAmounts,
+        address indexed beneficiary,
+        uint256 releaseTime
+    );
+
+    event AgreementClaimed(
+        uint256 agreementId,
+        DataTypes.AssetType assetType,
+        address indexed asset,
+        uint256[] tokenIdsOrAmounts,
+        address indexed beneficiary
+    );
+    mapping(uint256 => Agreement) public agreements;
+
+    uint248 public agreementCount;
```

Comment:
Should we update uint248 to uint256?

Reply:
it's `uint248` to save storage and get packed with the state variable after it into one slot

Reply:
Cool, understood.